python/GafferUI/AnimationEditor.py

```diff
@@ -868,4 +879,186 @@ def __checkBoxStateForTieMode( self, mode ) :
 		commonMode = self.getTieModeForSelectedKeys()
 		return None if commonMode is None else commonMode == mode
 
+# Private implementation - curve widget
+class _CurveWidget( GafferUI.GridContainer ) :
+
+	from collections import namedtuple
+	Connections = namedtuple( "Connections", ("color") )
+
+	def __init__( self ) :
+
+		GafferUI.GridContainer.__init__( self, spacing=4, borderWidth=4 )
+
+		# tool tips
+		colorToolTip = "# Color\n\nThe color of the currently selected curves.\n"
+
+		# create labels
+		colorLabel = GafferUI.Label( text="Color", toolTip=colorToolTip )
+
+		# create editors
+		self.__colorSwatch = GafferUI.ColorSwatch( useDisplayTransform=False, toolTip=colorToolTip )
+		self.__colorSwatch._qtWidget().setFixedHeight( 18 )
+		self.__colorInvalid = GafferUI.TextWidget( text="---", editable=False )
+		self.__colorInvalid.setEnabled( False )
+
+		# setup editor connections
+		self.__colorSwatch.buttonPressSignal().connect( Gaffer.WeakMethod( self.__swatchButtonPress ), scoped = False )
+		self.__colorSwatch.buttonReleaseSignal().connect( Gaffer.WeakMethod( self.__swatchButtonRelease ), scoped = False )
+		self.__colorSwatch.dragBeginSignal().connect( Gaffer.WeakMethod( self.__swatchDragBegin ), scoped = False )
+		self.__colorSwatch.dragEndSignal().connect( Gaffer.WeakMethod( self.__swatchDragEnd ), scoped = False )
+		self.__colorSwatch.contextMenuSignal().connect( Gaffer.WeakMethod( self.__swatchContextMenu ), scoped = False )
```

Comment:
It's a pity we can't use `ColorSwatchMetadataWidget` here (because it doesn't support multiple plugs at once). Let's not get into that now, but could you add a `## \todo` mentioning it please?

Reply:
done